Backend/Backend.WebServices/Controllers/EnterpriseController.cs

```diff
@@ -51,11 +51,26 @@ public IEnumerable<EnterpriseDTO> Get()
         /// Given an Id (and possibly a password), get details of a specific enterprise
         /// </summary>
         /// <param name="id">ID of the enterprise whose details will be returned</param>
-        /// <returns>A string representing an enterprise</returns>
-        public string Get(int id)
+        /// <param name="password">Password of the enterprise</param>
+        /// <returns>An EnterpriseDTO with the specified ID</returns>
+        public EnterpriseDTO Get(int id, string password = null)
```

Comment:
The password should not be a parameter in the GET request, instead we should get it from the headers. Can do this in the body of the method, something like 
```
string password = null;
if (Request.Headers.Any(x => "Authorization".Equals(x.Key)))
{
    password = Request.Headers.First(x => "Authorization".Equals(x.Key));
}
```